### PR TITLE
Added MRT support to vcGL

### PR DESCRIPTION
**Pull request by peterAdams1014**

### CODE REVIEW

**Comment by peterAdams1014** [comment]
This doesn't add MRT for Metal - we will resolve this later.

File: src/gl/vcFramebuffer.h [hunk 1]
```diff
@@ -16,7 +16,7 @@ enum vcFramebufferClearOperation
 
 struct vcFramebuffer;
 
-bool vcFramebuffer_Create(vcFramebuffer **ppFramebuffer, vcTexture *pTexture, vcTexture *pDepth = nullptr, uint32_t level = 0);
+bool vcFramebuffer_Create(vcFramebuffer **ppFramebuffer, vcTexture *pTexture, vcTexture *pDepth = nullptr, uint32_t level = 0, vcTexture *pAttachment2 = nullptr);
```
Comment (samsinsane) [inline src/gl/vcFramebuffer.h:19]: Should we perhaps consider creating a `vcFramebuffer_CreateAdv` which takes an array of attachments which indicate what kind of attachment they are and the `vcTexture*` to use?
```c++
enum vcFramebufferAttachmentType
{
  vcFAT_Colour,
  vcFAT_Depth
};

struct vcFramebufferAttachment
{
  vcTexture *pTexture;
  vcFramebufferAttachmentType type;
};
```

File: src/gl/opengl/vcFramebuffer.cpp [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 
-bool vcFramebuffer_Create(vcFramebuffer **ppFramebuffer, vcTexture *pTexture, vcTexture *pDepth /*= nullptr*/, uint32_t level /*= 0*/)
+bool vcFramebuffer_Create(vcFramebuffer **ppFramebuffer, vcTexture *pTexture, vcTexture *pDepth /*= nullptr*/, uint32_t level /*= 0*/, vcTexture *pAttachment2 /*= nullptr*/)
 {
   if (ppFramebuffer == nullptr || pTexture == nullptr)
     return false;
 
   udResult result = udR_Success;
-  static const GLenum DrawBuffers[] = { GL_COLOR_ATTACHMENT0 };
+  static const GLenum DrawBuffers[] = { GL_COLOR_ATTACHMENT0, GL_COLOR_ATTACHMENT1 };
```
Comment (samsinsane) [inline src/gl/opengl/vcFramebuffer.cpp:10]: I vaguely recall it being suggested that these should be done as `GL_COLOR_ATTACHMENT0 + N` instead - I think it was due to the other values not always being defined or something? It's possible I'm thinking of a different OpenGL enum set?

Comment (peterAdams1014) [reply src/gl/opengl/vcFramebuffer.cpp:10]: You are correct. I don't mind either way to be honest.

File: src/gl/vcFramebuffer.h (hunk 1 above)
Comment (peterAdams1014) [reply src/gl/vcFramebuffer.h:19]: Yeah sounds good - this change would be consistent with the vcTexture_CreateAdv() API. Probably won't make this change for this PR though.

Comment (samsinsane) [reply src/gl/vcFramebuffer.h:19]: > Probably won't make this change for this PR though.

Minor issue with going down this path is that `vcFramebuffer_Create` will need to have this change undone, and all usages moved to use `vcFramebuffer_CreateAdv`.

**Pull request merged**